### PR TITLE
Alignment to what has been chosen

### PR DESCRIPTION
**Pull request by omer-dayan**

### CODE REVIEW

File: cmd/job/submit/submit.go [hunk 1]
```diff
@@ -258,16 +259,25 @@ func (submitArgs *submitArgs) addCommonFlags(fbg flags.FlagsByGroups) {
 func (submitArgs *submitArgs) setCommonRun(cmd *cobra.Command, args []string, kubeClient *client.Client, clientset kubernetes.Interface, configValues *string) error {
 	util.SetLogLevel(global.LogLevel)
 	var name string
-	if nameParameter == "" && len(args) >= 1 {
+	if nameParameter != "" {
+		if len(args) > 0 || submitArgs.namePrefix != "" {
+			submitArgs.generateSuffix = false
```
Comment (yodarshafrir1) [inline cmd/job/submit/submit.go:264]: Isn't this the default value anyway?

File: cmd/job/submit/submit.go [hunk 2]
```diff
@@ -258,16 +259,25 @@ func (submitArgs *submitArgs) addCommonFlags(fbg flags.FlagsByGroups) {
...
+			log.Warn("Using the name provided with --name flag. Ignoring the other options")
+		}
+		name = nameParameter
+	} else if len(args) > 0 {
+		if submitArgs.namePrefix != "" {
+			log.Warn("Ignoring --job-name-prefix flag")
+			submitArgs.generateSuffix = false
+		}
+		log.Warn("Provide name without --name flag has been deprecated")
```
Comment (yodarshafrir1) [inline cmd/job/submit/submit.go:273]: cobra has a message for deprecated flags, i just ran: `runai submit --name blah`
`Flag --name has been deprecated, please use positional argument instead`
So I guess this is better:
`Submitting the job name as a positional argument has been deprecated, please use --name flag instead`.
 Also let's not use warn here?

File: cmd/job/submit/submit_runai.go [hunk 3]
```diff
@@ -29,16 +29,16 @@ import (
 const (
 	submitExamples = `
 # Start a Training job.
-runai submit train1 -i gcr.io/run-ai-demo/quickstart -g 1
+runai submit --name train1 -i gcr.io/run-ai-demo/quickstart -g 1
```
Comment (yodarshafrir1) [inline cmd/job/submit/submit_runai.go:32]: 🥇

File: pkg/workflow/workflow.go [hunk 4]
```diff
@@ -160,34 +160,33 @@ func getSmallestUnoccupiedIndex(configMaps []corev1.ConfigMap) int {
 	return len(configMaps)
 }
 
-func getConfigMapName(name string, index int) string {
-	if index == 0 {
+func getConfigMapName(name string, index int, generateSuffix bool) string {
+	if !generateSuffix {
```
Comment (yodarshafrir1) [inline pkg/workflow/workflow.go:164]: Didn't we say we want to start from index 0?

File: cmd/job/submit/submit.go [hunk 5]
```diff
@@ -258,16 +259,25 @@ func (submitArgs *submitArgs) addCommonFlags(fbg flags.FlagsByGroups) {
 func (submitArgs *submitArgs) setCommonRun(cmd *cobra.Command, args []string, kubeClient *client.Client, clientset kubernetes.Interface, configValues *string) error {
 	util.SetLogLevel(global.LogLevel)
 	var name string
-	if nameParameter == "" && len(args) >= 1 {
+	if nameParameter != "" {
+		if len(args) > 0 || submitArgs.namePrefix != "" {
+			submitArgs.generateSuffix = false
+			log.Warn("Using the name provided with --name flag. Ignoring the other options")
```
Comment (yodarshafrir1) [inline cmd/job/submit/submit.go:265]: Let's rephrase this

**Review by yodarshafrir1 — CHANGES_REQUESTED** [review]
Please see my notes

File: cmd/job/submit/submit.go (hunk 1 above)
Comment (omer-dayan) [reply cmd/job/submit/submit.go:264]: You sure right

File: pkg/workflow/workflow.go (hunk 4 above)
Comment (omer-dayan) [reply pkg/workflow/workflow.go:164]: Yes as you can see in the line 171 we call the method the first attempt with index 0

**Review by yodarshafrir1 — CHANGES_REQUESTED** [review]

**Review by yodarshafrir1 — APPROVED** [review]
💯

**Pull request merged**